hack/lib/init.sh

```diff
@@ -31,6 +31,10 @@ unset CDPATH
 # they can explicitly set GO111MODULE=on
 export GO111MODULE=off
 
+if [[ "${KUBE_PROVIDERLESS:-"off"}" == "on" ]]; then
+  export GOFLAGS=${GOFLAGS:-"-tags=providerless"}
+fi
```

Comment:
if there is GOFLAG sets this will not make anything, shouldn´t we try to append?

@liggitt or @benthelder or @thockin are the best persons to judge this, I´m always confused with the bash scripts in this folder

Reply:
cc @BenTheElder

Reply:
or maybe the job should set the GOFLAGS variable directly intead of using another env variable for this?

Reply:
@aojea Since this `KUBE_PROVIDERLESS` only going to be used by some CI jobs, i'll be making sure there is no GOFLAGS which can trip up the if condition. 

Also you can see that the flag works (and reduces the size of binaries) here:
https://github.com/kubernetes/kubernetes/pull/122595#issuecomment-1881376791

So we should be good

Reply:
We have existing code to merge with `GOFLAGS=-tags=foo` https://github.com/kubernetes/kubernetes/blob/4c82da4d9f2ddfb93a2b3528ed023863f66da486/hack/lib/golang.sh#L881

Preferably we should do this here as well?

Reply:
i wanted to keep it simple, but yea, adding it now.